packages/langium-railroad/src/grammar-railroad.ts

```diff
@@ -0,0 +1,125 @@
+/******************************************************************************
+ * Copyright 2023 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import { GrammarAST, findNameAssignment } from 'langium';
+import type { FakeSVG } from 'railroad-diagrams';
+import * as rr from 'railroad-diagrams';
+
+export const defaultCss = `
+svg.railroad-diagram {
+    background-color: hsl(30,20%,95%);
+}
+svg.railroad-diagram path {
+    stroke-width: 3;
+    stroke: black;
+    fill: rgba(0,0,0,0);
+}
+svg.railroad-diagram text {
+    font: bold 14px monospace;
+    text-anchor: middle;
+}
+svg.railroad-diagram text.label {
+    text-anchor: start;
+}
+svg.railroad-diagram text.comment {
+    font: italic 12px monospace;
+}
+svg.railroad-diagram rect {
+    stroke-width: 3;
+    stroke: black;
+    fill: hsl(120,100%,90%);
+}
+`.trim();
+
+export interface GrammarDiagramOptions {
+    css?: string;
+    javascript?: string;
+}
+
+export function createGrammarDiagramHtml(grammar: GrammarAST.Grammar, options?: GrammarDiagramOptions): string {
+    let text = `<!DOCTYPE HTML>
+<html>
+<head>
+<style>
+${defaultCss}
+</style>${options?.css ? `
+<style>
+${options.css}
+</style>` : ''}
+${options?.javascript ? `
+<script>
+${options.javascript}
+</script>` : ''}
+</head>
+<body>
+`;
+    text += createGrammarDiagram(grammar);
+    text += `</body>
+</html>`;
+
+    return text;
+}
+
+export function createGrammarDiagram(grammar: GrammarAST.Grammar): string {
+    const nonTerminals = grammar.rules.filter(GrammarAST.isParserRule);
+    let text = '';
```

Comment:
Are rule fragments a problem? Are they inlined or normal rules?

Reply:
They are treated as being normal rules. I'm wondering whether it makes sense to inline them.

Reply:
I think it is fine to treat them as normal rules.